client/src/api.ts

```diff
@@ -98,7 +99,7 @@ export async function continueSession(props: {
       `${DIALOG_ENDPOINT}${props.lesson}/session`,
       {
         sessionInfo: props.session,
-        message: props.outboundChat,
+        message: props.outboundChat.slice(0, MAX_INPUT_LENGTH),
```

Comment:
Do you need to slice the message every time you fetch the session? `props.outboundChat.slice(0, MAX_INPUT_LENGTH)` If so, that's fine but it looks a little bit weird because you might need it complete in other cases.